pkg/commands/command_helper.go

```diff
@@ -25,6 +31,8 @@ type CommandHelper struct {
 
 	objPrinter k8s.ObjectPrinter
 	strBuilder strings.Builder
+
+	typeToGVK map[reflect.Type]schema.GroupVersionKind
```

Comment:
This reflection now kinda makes me a little worried just because if reflection fails (and I know it shouldn't because we should only be working with types that we expect) we'll have a `panic` somewhere.

I kinda wonder if we need to use the `k8s.io/apimachinery/pkg/runtime.Object` interface for the `github.com/pivotal/build-service-cli/pkg/k8s.ObjectPrinter.PrintObj` method parameter type.

What if we create our own interface which "extends" `k8s.io/apimachinery/pkg/runtime.Object` and includes a `Name() string` method because all k8s resources will implement that. Then we can just used `Name()` for the hash table keys instead of `reflect.Type`

Reply:
> This reflection now kinda makes me a little worried just because if reflection fails (and I know it shouldn't because we should only be working with types that we expect) we'll have a `panic` somewhere.

Shouldn't cause a panic, but errors out with explicit info.... because it is checked for in the PrintObj(...)
```
nGVK, ok := ch.typeToGVK[reflect.TypeOf(obj)]
if !ok {
	return errors.Errorf("failed to print object. unknown type %q", reflect.TypeOf(obj))
}
```

> I kinda wonder if we need to use the `k8s.io/apimachinery/pkg/runtime.Object` interface for the `github.com/pivotal/build-service-cli/pkg/k8s.ObjectPrinter.PrintObj` method parameter type.
> 
> What if we create our own interface which "extends" `k8s.io/apimachinery/pkg/runtime.Object` and includes a `Name() string` method because all k8s resources will implement that. 
> Then we can just used `Name()` for the hash table keys instead of `reflect.Type`

I do like what you have suggested. I did take a similar approach (partly) for the `func SetLastAppliedCfg(obj Annotatable) error` by using the Annotatable interface
but did not extend the k8s.io/apimachinery/pkg/runtime.Object - will make that change

Not sure, if we can actually skip the reflection bit though. What we need is not the Name, but the Kind and Version values. The TypeMeta fields were empty after fetching an Object, specifically the ServiceAccount for secret create
```
serviceAccount, err := cs.K8sClient.CoreV1().ServiceAccounts(cs.Namespace).Get("default", metav1.GetOptions{})
```

I did google around for this and seemed like a known thing with `client-go`. Will look around more.

Reply:
Ah, I see. That is so annoying! (`ServiceAccount` having missing `TypeMeta`) But it guess that's what this PR is all about!

Reply:
also caters for our CustomResource object creation... we don't set TypeMeta in most places (I believe)